doc/en/user/source/rest/api/stylegenerator.rst

```diff
@@ -0,0 +1,125 @@
+.. _rest_api_stylegenerator:
+
+Style Generator
+===============
+
+The Style Generator endpoint is used for generating default styles. Styles generated using this endpoint may be added to geoserver using the :ref:`styles <rest_api_styles>` endpoint.
+
+``/stylegenerator[.<format>]``
+------------------------------
```

Comment:
The best idea might be to add `generate` and `type` parameters to the styles API, such that:
* POST `/styles?name=newstyle&generate=true&type=point` creates a new style named "newstyle", generated using the point template
* PUT `/styles/existingstyle?generate=true&type=point` replaces the existing style with a generated style.

I do have a couple concerns about this approach however:
* Unlike other POST/PUT requests against the styles API, these would not expect a form upload. Is this likely to cause issues? (I am not really familiar enough with how our REST API works to determine if this could be a problem)
* This is extending an existing endpoint with functionality that can easily wipe out an the style; seems a bit risky (althoug a blank form upload would probably do the same thing, so maybe not an issue).

@aaime Does this method for exposing the style generator using REST seem better then my current implementation that adds a new GET endpoint? Do you think an empty POST/PUT will cause problems?